firedrake/pointwise_operators.py

```diff
@@ -0,0 +1,485 @@
+from abc import ABCMeta, abstractmethod
+from functools import partial
+import types
+from ufl.core.external_operator import ExternalOperator
+from ufl.core.expr import Expr
+from ufl.algorithms.apply_derivatives import VariableRuleset
+from ufl.constantvalue import as_ufl
+from firedrake.function import Function
+from firedrake import utils, functionspaceimpl
+from pyop2.datatypes import ScalarType
+from ufl.log import error
+import sympy as sp
+import numpy as np
+from scipy import optimize
+
+
+class AbstractPointwiseOperator(Function, ExternalOperator, metaclass=ABCMeta):
+    r"""Asbtract base class from wich stem from all the firedrake practical implementation of the
```

Comment:
also "which"

Reply:
the second "from" is spurious
Firedrake has a capital F
implementations (plural)

Reply:
Oops, I changed that.